pkg/apis/networking/types.go

```diff
@@ -362,3 +405,39 @@ type IngressBackend struct {
 	// Specifies the port of the referenced service.
 	ServicePort intstr.IntOrString
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// IngressClass represents the class of the Ingress, referenced by the Ingress
+// Spec.
+type IngressClass struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	// Controller is responsible for handling this class. This should be
+	// specified as a domain-prefixed path, e.g. "acme.io/ingress-controller".
+	// This allows for different "flavors" that are controlled by the same
+	// controller. For example, you may have different Parameters for the same
+	// implementing controller.
+	Controller string
+
+	// Parameters is a link to a custom resource configuration for the
+	// controller. This is optional if the controller does not require extra
+	// parameters.
+	// +optional
+	Parameters *api.ObjectReference
```

Comment:
Should these (Controller and Parameters) be nested under a `Spec` struct?

Reply:
Good catch, thanks! Although this wasn't specified in the KEP, I think it makes sense to move these under a `Spec` struct, will update.

Reply:
My concern is similar to @hbagdi . When I took a stab at this I called out that the KEP example may have been wrong but I didn't know if it should be something that is nested in it's own spec or within the existing i.e https://github.com/kubernetes/kubernetes/pull/83467/files

Reply:
Updated in the KEP and now here - we now have a Spec to make room for the potential addition of Status in the future.